google-oauth-client/src/main/java/com/google/api/client/auth/oauth2/AuthorizationCodeFlow.java

```diff
@@ -156,6 +184,7 @@ protected AuthorizationCodeFlow(Builder builder) {
     clientId = Preconditions.checkNotNull(builder.clientId);
     authorizationServerEncodedUrl =
         Preconditions.checkNotNull(builder.authorizationServerEncodedUrl);
+    additionalParameters = Preconditions.checkNotNull(builder.additionalParameters);    
```

Comment:
no need of null check here because it is an optional thing. You are checking for null before using it.

Reply:
Ah! Removed precondition check.